### PR TITLE
Introduce EnclaveID as identifier

### PR DESCRIPTION
**Pull request by BedrockSquirrel**
### Why this change is needed

EnclaveID will be the identifier for enclaves that join the protocol.

It is calculated from the enclave's public key generated when it first starts up.

### What changes were made as part of this PR

Add method to provide EnclaveID to the host. Have the guardian request the EnclaveID before starting its main loop and use it in its logging (this will be necessary when a host has more than one guardian).

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/ten-protocol/ten-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: go/enclave/enclave.go [hunk 1]
```diff
@@ -87,6 +87,7 @@ type enclaveImpl struct {
 
 	enclaveKey    *ecdsa.PrivateKey // this is a key specific to this enclave, which is included in the Attestation. Used for signing rollups and for encryption of the shared secret.
```
Comment (tudor-malene) [inline go/enclave/enclave.go:88]: feels like we should group these fields and the logic into a struct that only has that responsibility.

**Review by tudor-malene — COMMENTED** [review]
Looks good.
Minor comments and a small refactoring suggestion

File: go/enclave/enclave.go (hunk 1 above)
Comment (BedrockSquirrel) [reply go/enclave/enclave.go]: Yeah good shout. Is the new commit the sort of thing you had in mind?

**Review by tudor-malene — APPROVED** [review]
lgtm

**Pull request merged**